Test/QuickCheck/Modifiers.hs

```diff
@@ -343,4 +355,100 @@ instance (Arbitrary a, ShrinkState s a) => Arbitrary (Shrinking s a) where
 #endif /* NO_MULTI_PARAM_TYPE_CLASSES */
 
 --------------------------------------------------------------------------
+-- | @ASCII@.
+newtype ASCII = ASCII {getASCII :: Char}
+  deriving ( Eq, Ord, Show, Read )
+
+asciiGen :: Gen Char
+asciiGen = choose ('\0', '\127')
+
+instance Arbitrary ASCII where
+  arbitrary = ASCII `fmap` asciiGen
+
+--------------------------------------------------------------------------
+-- | @Latin1@: latin1 Char, same Arbitrary as Char itself.
+newtype Latin1 = Latin1 {getLatin1 :: Char}
+  deriving ( Eq, Ord, Show, Read )
+
+instance Arbitrary Latin1 where
+  arbitrary = Latin1 `fmap` arbitrary
+  shrink (Latin1 c) = Latin1 `fmap` shrink c
+
+--------------------------------------------------------------------------
+-- | @Unicode@: yields unicode Char.
+newtype Unicode = Unicode {getUnicode :: Char}
+  deriving ( Eq, Ord, Show, Read )
+
+unicodeGen' :: Gen Char
+unicodeGen' = oneof [choose (minBound, '\55296'), -- Skip high and low surrogates
+                     choose ('\57344', maxBound)]
+
```

Comment:
Yup, but the numbers would be awkward... I'll calculate them when I get back to my laptop